File: examples/javascript/quickstart.ts
/**
 * CoreLink JS/TS WASM quickstart — @corelink/client.
 *
 * Requires:
 *   npm install @corelink/client
 *   export CORELINK_PAT=<your-pat>
 *
 * Run:
 *   npx ts-node quickstart.ts
 *   # or
 *   node --experimental-vm-modules dist/quickstart.js
 */

import { CoreLinkClient } from "@corelink/client";

async function main(): Promise<void> {
  const pat = process.env.CORELINK_PAT;
  if (!pat) {
    console.error("ERROR: CORELINK_PAT environment variable not set");
    process.exit(1);
  }

  // clientVerify defaults to true per CTRL-CAS-002.
  // Pass clientVerify: false to opt out (warning logged).
  const client = new CoreLinkClient({
    pat,
    tenantId: "acme-corp",
    // clientVerify: true  // default; shown for clarity
  });

  console.log(`_clientVerifyEnabled: ${client._clientVerifyEnabled}`);

  // Put
  const data = new TextEncoder().encode("hello from CoreLink JS/TS SDK");
  const digest = await client.put(data);
  console.log(`Uploaded:  blake3:${digest}`);

  // Get (client-verify enforced by Rust single truth via WASM)
  const downloaded = await client.get(digest);
  console.log(`Downloaded: ${downloaded.length} bytes`);

  // Stat
  const stat = await client.stat(digest);
  console.log(
    `Stat:      digest=${stat.digest.slice(0, 8)}... ` +
    `size=${stat.sizeBytes} exists=${stat.exists}`
  );

  console.log("OK — JS/TS SDK quickstart complete.");
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: examples/python/quickstart.py
#!/usr/bin/env python3
"""
CoreLink Python SDK quickstart — corelink-py.

Requires:
    pip install corelink-py

Environment:
    CORELINK_PAT=<your-pat>

Usage:
    python quickstart.py
"""

import asyncio
import os
import sys

from corelink import CoreLinkClient  # type: ignore[import]


async def main() -> None:
    """Demonstrate put/get/stat with client-verify default-on."""
    pat = os.environ.get("CORELINK_PAT", "")
    if not pat:
        print("ERROR: CORELINK_PAT environment variable not set", file=sys.stderr)
        sys.exit(1)

    # client_verify defaults to True per CTRL-CAS-002.
    # Pass client_verify=False to opt out (warning logged).
    client = CoreLinkClient(
        pat=pat,
        tenant_id="acme-corp",
        # client_verify=True  # default; shown for clarity
    )

    print(f"client_verify_enabled: {client._inner_client_verify_enabled}")

    # Put
    data = b"hello from CoreLink Python SDK"
    digest = client.put(data)
    print(f"Uploaded:  blake3:{digest}")

    # Get (client-verify enforced by Rust single truth)
    downloaded = client.get(digest)
    print(f"Downloaded: {len(downloaded)} bytes")
    assert downloaded == data, "data round-trip mismatch"

    # Stat
    stat = client.stat(digest)
    print(f"Stat:      digest={stat.digest[:8]}... size={stat.size_bytes} exists={stat.exists}")

    print("OK — Python SDK quickstart complete.")


if __name__ == "__main__":
    asyncio.run(main())
